var/spack/repos/builtin/packages/ecflow/package.py

```diff
@@ -73,23 +75,41 @@ def patch(self):
             "Pyext/CMakeLists.txt",
         )
 
+    @when("+ssl ^openssl~shared")
+    def setup_build_environment(self, env):
+        env.set("LIBS", self.spec["zlib"].libs.search_flags)
+
     def cmake_args(self):
-        boost_lib = self.spec["boost"].prefix.lib
-        return [
+        spec = self.spec
+        boost_lib = spec["boost"].prefix.lib
+        args = [
             self.define("Boost_PYTHON_LIBRARY_RELEASE", boost_lib),
             self.define_from_variant("ENABLE_UI", "ui"),
             self.define_from_variant("ENABLE_GUI", "ui"),
             self.define_from_variant("ENABLE_SSL", "ssl"),
             # https://jira.ecmwf.int/browse/SUP-2641#comment-208943
             self.define_from_variant("ENABLE_STATIC_BOOST_LIBS", "static_boost"),
-            self.define("Python3_EXECUTABLE", self.spec["python"].package.command),
-            self.define("BOOST_ROOT", self.spec["boost"].prefix),
+            self.define("Python3_EXECUTABLE", spec["python"].package.command),
+            self.define("BOOST_ROOT", spec["boost"].prefix),
+            self.define("CMAKE_POSITION_INDEPENDENT_CODE", "ON"),
```

Comment:
Should this be only if `~shared`?

Reply:
I just added a pic variant to make it more consistent with other packages.